coreneuron/io/reports/nrnreport.hpp

```diff
@@ -101,7 +102,13 @@ struct ReportConfiguration {
     double stop;                          // stop time of report
     int num_gids;                         // total number of gids
     int buffer_size;                      // hint on buffer size used for this report
-    std::set<int> target;                 // list of gids for this report
+    std::vector<int> target;              // list of gids for this report
+
+    ~ReportConfiguration() {
+        // Force cleanup of the std::vector and release the memory
+        std::vector<int>().swap(target);
+        malloc_trim(0);
```

Comment:
bit confused by this - I kind of see why this is done. But I wonder if this type of pattern we really need to do in C++ part (especially if we know we are releasing a large vector). Thoughts? 🤔

Reply:
I did this because otherwise I didnt see the memory being released...but maybe you are right and we should let the system figure it out

Reply:
I'm opposed to scattering things like this throughout the code...

Reply:
I strongly agree with @olupton , we should not be doing this left and right. I can see how at one place at the end of the model building in neurodamus that made sense, but this should not become the norm.